router/core/src/main/scala/io/buoyant/router/RoutingFactory.scala

```diff
@@ -165,7 +167,11 @@ class RoutingFactory[Req, Rsp](
             Future.exception(UnknownDst(req0, unidentified.reason))
           case Throw(e) =>
             Annotations.Failure.Identification.record(e.getMessage)
-            Future.exception(UnknownDst(req0, e.getMessage))
+            e match {
+              case e: ResponseException[Rsp] => Future.exception(e)
```

Comment:
I could use some feedback here. Without this change, the ResponseException never makes it to the ErrorResponder/ErrorResetter. But with this change, the exception also gets propagated to the default monitor:

```
Exception propagated to the default monitor (upstream address: /172.17.0.1:50971, downstream address: n/a, label: 0.0.0.0/80).
io.buoyant.linkerd.protocol.http.ErrorResponder$HttpResponseException
```
Makes me doubt that ErrorResponder is the right place in the stack to be handling these kinds of responses. Thoughts?